Support Anthropics OAuth as new provider

mikaelliljedahl opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Provider system with Anthropic OAuth login (PKCE) and OpenRouter adapter, provider selection/settings UI, browser-based login dialogs, encrypted token storage, and provider-aware model selection.

- **Refactor**
  - Pluggable provider interfaces, unified streaming/non-streaming chat bridging, provider-aware startup/configuration, and UI updated to show/manage current provider and models.

- **Tests**
  - Expanded unit/integration tests covering providers, auth, token store, PKCE, message conversion, streaming, and provider factory.

- **Chores**
  - Config validation, atomic config writes, launch settings and dependency updates.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mikaelliljedahl committed:
added provider selectiong anthropics (not wroking)

mikaelliljedahl committed:
WIP: Save local changes before rebase

mikaelliljedahl committed:
works with claude OAuth provider now by stripping down system prompt nad moving it to first user prompt

mikaelliljedahl committed:
remove local files

mikaelliljedahl committed:
fix build errors

mikaelliljedahl committed:
removed excessive debug logging

mikaelliljedahl committed:
remove debug logging

mikaelliljedahl committed:
fix suggestions from code rabbit

mikaelliljedahl committed:
merge from master + fix build errors

mikaelliljedahl committed:
fix some warnings

mikaelliljedahl committed:
coderabbiit suggestions

mikaelliljedahl committed:
fixed som build warnings and aligned coding style for the new anthropic stuff

mikaelliljedahl committed:
more null warning fixes

mikaelliljedahl committed:
added better instructions

mikaelliljedahl commented:
Only tested using the Claude subscription, not using openrouter. I only tested simple command like list contents of this folder, what does the project do.

mikaelliljedahl committed:
fix backwards compatability

mikaelliljedahl committed:
fix tests pass

mikaelliljedahl committed:
merged from master

mikaelliljedahl committed:
fix merge with master

mikaelliljedahl committed:
fix provider jumped back to openrouter

mikaelliljedahl committed:
linux test failed - fix

mikaelliljedahl committed:
fix test on linux/mac

mikaelliljedahl committed:
fix tests on linux, part 1

mikaelliljedahl committed:
fix tests on linux (wsl)  part 2

mikaelliljedahl committed:
removed default 3.5 sonnet model. Fix bug when provider was reset after restart

mikaelliljedahl committed:
avoid these tests run concurrently failing the test in WSL

mikaelliljedahl committed:
remove hard coded models (thanks coderabbit)

mikaelliljedahl committed:
moved model mappings for antropics to a json file

mikaelliljedahl committed:
Use streaming with Claude provider

mikaelliljedahl committed:
fix UI for Claude Auth in WSL

mikaelliljedahl committed:
fix first time setup with Anthropics

xyOz-dev commented on Providers/Anthropic/AnthropicClient.cs:
Please review all suggestions by yourself, remove anything outdated. Do a new full review and only point out potential issues, remove everything thats a refactor suggestions, simply issues that may affect users.

mikaelliljedahl committed:
do not mess with real tokenstore during unit testing. Fix temperature check and TopP check (not using both with Opus)

mikaelliljedahl committed:
replace Console.WriteLine with System.Diagnostics.Debug.
use InfiniteTimeSpan for the sse connection to avoid timeout for long running agents

mikaelliljedahl committed:
skip throwing if topP is set
